airgun/views/hostcollection.py

```diff
@@ -57,9 +73,236 @@ def is_displayed(self):
+                else:
+                    return self.list_hosts.read()
+
+            def click(self):
+                if self.add_hosts.is_displayed:
+                    # This click should open hosts>resources>AddTab
+                    return self.add_hosts.click()
+                else:
+                    # This click should open hosts>resources>ListRemoveTab
+                    return self.list_hosts.click()
 
     @View.nested
     class hosts(SatTab):
         TAB_NAME = 'Hosts'
 
         resources = View.nested(AddRemoveResourcesView)
+
+
+class HostCollectionPackageContentRadioGroup(RadioGroup):
+    """Handle an HTML non normalized Radio group according to the current
+    architecture.
+    Note: This is a temporary solution, a fix will be issued upstream,
+        when the fix will be available downstream we should replace the
+        implementation with RadioGroup.
+    Example html representation::
+
+        <div >
+            <input type="radio" id="package" ...>
+            <label>Package</label>
+            <input type="radio" id="package_group" ...>
+            <label>Package Group</label>
+      """
+    # a mapping between button name and the id, see the implementation in
+    # HostsAssignOrganization and HostsAssignLocation hereafter.
+    buttons_name_id_map = {
+        'Package': 'package',
+        'Package Group': 'package_group',
+    }
+
+    def get_input_by_name(self, name):
+        input_id = self.buttons_name_id_map.get(name)
+        if input_id:
+            return self.browser.wait_for_element(
+                ".//input[@id='{0}']".format(self.buttons_name_id_map[name])
+            )
+        return None
+
+    @property
+    def selected(self):
+        """Return the name of the button that is currently selected."""
+        for name in self.buttons_name_id_map.keys():
+            btn = self.get_input_by_name(name)
+            if btn and btn.get_attribute('checked') is not None:
+                return name
+        else:
+            raise ValueError(
+                "Whether no radio button is selected or proper attribute "
+                "should be added to framework"
+            )
+
+    def select(self, name):
+        """Select specific radio button in the group"""
+        if self.selected != name:
+            btn = self.get_input_by_name(name)
+            if btn:
+                btn.click()
+                return True
+        return False
+
+    def read(self):
+        """Wrap method according to architecture"""
+        return self.selected
+
+    def fill(self, name):
+        """Wrap method according to architecture"""
+        return self.select(name)
+
+
+class HostCollectionManagePackagesView(BaseLoggedInView):
+    title = Text("//h4[contains(., 'Update Packages')]")
+    update_all = ActionsDropdown(
+        "//button[contains(@class, 'btn')][contains(@ng-click, 'update all')]"
+        "/ancestor::span[contains(@class, 'input-group')]"
```

Comment:
mind charing html code snippet? I feel like there should be simpler way to locate these elements. From the first glance - it's smth like `//span[contains(@class, 'input-group')][button[..]]` to me, isn't it?